src/Files.App/Views/Pages/PropertiesGeneral.xaml.cs

```diff
@@ -1,113 +1,131 @@
-                {
-                    foreach (var fileOrFolder in combinedProps.List)
-                    {
-                        await App.Window.DispatcherQueue.EnqueueAsync(() => UIFilesystemHelpers.SetHiddenAttributeItem(fileOrFolder, ViewModel.IsHidden, AppInstance.SlimContentPage.ItemManipulationModel));
-                    }
-                }
-                return true;
-            }
-            else
-            {
-                // Handle the visibility attribute for a single file
-                if (AppInstance?.SlimContentPage?.ItemManipulationModel != null) // null on homepage
-                {
-                    await App.Window.DispatcherQueue.EnqueueAsync(() => UIFilesystemHelpers.SetHiddenAttributeItem(item, ViewModel.IsHidden, AppInstance.SlimContentPage.ItemManipulationModel));
-                }
-
-                ViewModel.ItemName = ItemFileName.Text; // Make sure Name is updated
-                if (!string.IsNullOrWhiteSpace(ViewModel.ItemName) && ViewModel.OriginalItemName != ViewModel.ItemName)
-                {
-                    return await App.Window.DispatcherQueue.EnqueueAsync(() => UIFilesystemHelpers.RenameFileItemAsync(item,
-                          ViewModel.ItemName,
-                          AppInstance));
-                }
-                return true;
-            }
-
-            return false;
-        }
-
-        public override void Dispose()
-        {
-        }
-
-        private void DiskCleanupButton_Click(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
-        {
-            if (BaseProperties is DriveProperties driveProps)
-            {
-                var drive = driveProps.Drive;
-
-                StorageSenseHelper.OpenStorageSense(drive.Path);
-            }
-        }
-    }
-}
\ No newline at end of file
+	public sealed partial class PropertiesGeneral : PropertiesTab
+	{
+		private readonly Regex letterRegex = new(@"\s*\(\w:\)$");
+
+		public PropertiesGeneral() => InitializeComponent();
+
+		public override async Task<bool> SaveChangesAsync(ListedItem item)
+		{
+			ViewModel.ItemName = ItemFileName.Text; // Make sure Name is updated
+
+			string newName = ViewModel.ItemName;
+			string oldName = ViewModel.OriginalItemName;
+			bool hasNewName = !string.IsNullOrWhiteSpace(newName) && newName != oldName;
+
+			var fsVM = AppInstance.FilesystemViewModel;
+			var itemMM = AppInstance?.SlimContentPage?.ItemManipulationModel;
+
+			if (BaseProperties is DriveProperties driveProps)
+			{
+				if (!hasNewName || fsVM is null)
+					return false;
+
+				var drive = driveProps.Drive;
+				newName = letterRegex.Replace(newName, string.Empty); // Remove "(C:)" from the new label
+
+				Win32API.SetVolumeLabel(drive.Path, newName);
+				_ = App.Window.DispatcherQueue.EnqueueAsync(async () =>
+				{
+					await drive.UpdateLabelAsync();
+					await fsVM.SetWorkingDirectoryAsync(drive.Path);
+				});
+				return true;
```

Comment:
Maybe extract the `if` instructions into a method to make `SaveChangesAsync ` smaller and easier to read?